### ISSUE TITLE
Tasmota in Tywe3s board - curtain motor

### ISSUE THREAD

**Issue by hecarrara** [arendst/Sonoff-Tasmota#4434]
I bought a curtain motor and found inside the TYWE3S board. Anyone already use this board with tasmota for curtain motors? I only saw with smartplugs.

![whatsapp image 2018-11-23 at 19 44 22](https://user-images.githubusercontent.com/44953407/48961102-324f8480-ef59-11e8-9825-06e79e77e4ca.jpeg)
![whatsapp image 2018-11-23 at 19 44 21](https://user-images.githubusercontent.com/44953407/48961103-324f8480-ef59-11e8-95b2-a38249f999dc.jpeg)
![whatsapp image 2018-11-23 at 19 44 23](https://user-images.githubusercontent.com/44953407/48961104-324f8480-ef59-11e8-8849-47e7f6760ba2.jpeg)

I upload the tasmota firmware successfully but I'm trying to discover which gpios are the relays, buttons, led, etc. 
Excuse me if I say some absurd, I'm don't have technical knowledge...

**Comment by ascillato** [comment]
I don't know if Tasmota works whithout adaptations in that device. So, please, set the module to generic and put your gpios as relays and try to turn each one to see if something happens and write down your findings. If the ESP8266 is controlling directly the relays, you will be able to use Tasmota. If you can't turn your relays, means that there is another MCU that controls them. If that is the case, some reverse engineering needs to be done to know the actual protocol it uses.

Please, tell us is you could control your relays using generic module. Thanks

**Comment by digiblur** [comment]
What motor is that and where did you get it from?

**Comment by hecarrara** [comment]
Here is my current discovery:
With this config, on toggle button 4, the motor turns on and then off automatically (video).
I tried to leave **only** GPIO13 (as button 4) config as relay, but then the motor doesn't respond.
![capture2](https://user-images.githubusercontent.com/44953407/48974178-cf80ea80-f037-11e8-92fc-7bd567a24561.PNG)
![capture](https://user-images.githubusercontent.com/44953407/48974180-d0198100-f037-11e8-820b-3315694ec4da.PNG)

This motor also has two 44e402 hall sensors to report the rotation direction.

**Comment by ascillato2** [comment]
Hi, any news on this? Could you make it work properly?

**Comment by ascillato2** [comment]
Up to now, motor positions are not supported in Tasmota. That is why it is still open the issue https://github.com/arendst/Sonoff-Tasmota/issues/288

**Comment by ascillato2** [comment]
Hi, any news on this? Could you make it work properly?

**Comment by hecarrara** [comment]
Not yet. I just have time to work on this on the weekends. Unfortunelly.

**Comment by ascillato2** [comment]
Ok, no problem at all. Please, if you need any help or assistant, jusk ask here or in the Tasmota Chat

Thanks for sharing this.

**Comment by hecarrara** [comment]
Hello!! I had some good discovery yesterday.
Basicly the system has 2 relays, one Differential Comparator (for detecting when the curtain is stuck at the end, the current motor increases and the eletronic limit is set), 2 44e402 hall sensors, 1 learning button for rx480-s 433 signal,  one led, 2 rs1 and rs2 rj11 conectors.
So, besides the Tywe3s board, there is 2 other STC 15W404AS controllers on the board (I will call PIC 1 and PIC2)
One is taking care of the relays, hall sensors and the LM393 Differential Comparator.
The other is taking care of rx-480-s 433 signal, the led, the learn button.
I bought another curtain motor on 11.11 sale from aliexpress, this time is DOOYA brand, but is exactly the same system. Even in the INSMA motor the PIC controllers was scratched so I just could discover their types by looking in the DOOYA motor.
Here is the connections I found:
PIC 1 - PIN 8 - GND
PIC 1 - PIN 6 VCC
PIC 1 - PIN 9 RX
PIC 1 - PIN 10 TX
PIC 1 - PIN 7 - RELAY 
PIC 1 - PIN 5 - RELAY
PIC 1 - PIN 12 - HALL 2 -> PIC 2 - PIN 6
PIC 1 - PIN 13 - HALL 1
PIC 1 PIN 4 -> GPIO12 TYWE3S -> PIC 2 -> PIN 7
GPIO13 -> PIC 2 PIN 1
PIC 1 PIN 11 -> LM393
PIC 2 - PIN 11 -> BLUE LED
PIC 2 PIN 13 - PIC 2 PIN 5
PIC 2 - PIN 4 - LEARN BUTTON
PIC 2 - PIN 14 - GPIO 4
PIC 2 - PIN 12 - RX-480S RF 433 SIGNAL

That was the good news. *I'm not 100% sure, haven't test with tasmota yet.*
Now the bad news is that the tin pin I soldered has removed the VCC slot on the TYWE3S board. I will try to connect yet.

Below pictures of DOOYA motor.

![whatsapp image 2018-12-02 at 09 31 08](https://user-images.githubusercontent.com/44953407/49339430-b28c7e80-f618-11e8-8e50-d70fa1ad662d.jpeg)
![whatsapp image 2018-12-02 at 09 31 13](https://user-images.githubusercontent.com/44953407/49339431-b28c7e80-f618-11e8-8566-b9e19f222bd3.jpeg)
![whatsapp image 2018-12-02 at 09 31 081](https://user-images.githubusercontent.com/44953407/49339432-b28c7e80-f618-11e8-8db8-d842689df954.jpg)
